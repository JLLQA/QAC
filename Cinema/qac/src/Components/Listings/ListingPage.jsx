import axios from "axios";
import { useEffect, useState } from "react";
import Grid from "./ListingDisplay/Grid";
import Navbar from "../Multipage/Navbar/Navbar";

const ListingPage = () => {
    
    const [data, setData] = useState([]);
    const [isLoaded, setIsLoaded] = useState(false);

    useEffect(() => {
        axios({
            method: "Get",
            url: "http://127.0.0.1:5000/movies",
            headers: { "Access-Control-Allow-Origin": "*"}
        })
            .then((resp) => {
                console.log(resp);
                setData(resp.data);
                setIsLoaded(true);
            })
            .catch((err) => {
                console.log(err.message);
                setIsLoaded(true);
            });
    }, []);

    if (isLoaded) {
        return (
            <div>
                <Navbar />
                <Grid data={data} />
            </div>
        );
    } else {
        return (
            <div>
                <Navbar />
                <h1>Loading...</h1>
            </div>
        );
    }
}

export default ListingPage;