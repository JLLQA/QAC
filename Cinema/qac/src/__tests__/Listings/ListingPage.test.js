import '@testing-library/jest-dom';
import ListingPage from '../../../src/Components/Listings/ListingPage';
import renderer from 'react-test-renderer';

describe ("Listing page testing", () => {

    test('It matches snapshot', () => {
        const tree = renderer.create(<ListingPage/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

})