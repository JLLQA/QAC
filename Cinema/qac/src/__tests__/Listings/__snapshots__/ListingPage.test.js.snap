// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing page testing It matches snapshot 1`] = `
<div
  className="container-fluid"
  id="dropped-box"
>
  <h1>
    Loading...
  </h1>
</div>
`;
